src/ManagedArray.hpp

```diff
@@ -297,6 +297,32 @@ makeManagedArray(
   return array;
 }
 
+/*!
+ * \brief Create a copy of the given ManagedArray with a single allocation in the active
+ *  space of the given array.
+ *
+ * \param array The ManagedArray to copy.
+ *
+ * \tparam T Type of the raw data.
+ *
+ * \return A copy of the given ManagedArray.
+ */
+template <typename T>
+ManagedArray<T>
+deepCopy(
+    ManagedArray<T> const& array)
+{
+  T* data_ptr = array.getActivePointer();
+  
+  ArrayManager* manager = ArrayManager::getInstance();
+  
+  PointerRecord const* record = manager->getPointerRecord(data_ptr);
+  
+  PointerRecord* copy_record = manager->copyRecord(record);
```

Comment:
It make more sense to call this `copyFromRecord` or `deepCopyRecord` - otherwise it sounds like it just copies the record and not the data.

Reply:
Done.